app/contexts/registration_context.rb

```diff
@@ -0,0 +1,43 @@
+class RegistrationContext < ApplicationContext
```

Comment:
I broke this logic out from UserController because there's a lot happening here, and probably more eventually.

The UserController#create action was already getting a little bloated, and I think it's worth breaking out into a dedicated bit of code.

I tried to create a healthy, usable interface, something that is pretty inspired by railsisms (with ActiveModel::Errors), is testable, and doesn't bloat the controller.

Comments welcome!